src/components/vanilla/charts/TableChart/index.tsx

```diff
@@ -108,6 +109,12 @@ export default (props: Props) => {
     [meta, setMeta],
   );
 
+  useEffect(() => {
+    if (props.results?.data?.length) {
+      setHasNextPage(props.limit ? props.results.data.length >= props.limit : false);
```

Comment:
same here. if the length is equal to limit then we should not set the state to true.